src/units.jl

```diff
@@ -126,12 +127,12 @@ Parse a string containing an expression of units and return the
 corresponding `Quantity` object with `Float64` value. For example,
 `uparse("m/s")` would be parsed to `Quantity(1.0, length=1, time=-1)`.
 """
-function uparse(s::AbstractString)
-    return as_quantity(eval(Meta.parse(s)))::Quantity{DEFAULT_VALUE_TYPE,DEFAULT_DIM_TYPE}
+function uparse(s::AbstractString)::Quantity{LazyFloat64,DEFAULT_DIM_TYPE}
```

Comment:
This ensures that any unit is converted to a LazyFloat64 at the end. So even `u"1.0"` would become a LazyFloat64.